Jenkinsfile

```diff
@@ -165,9 +165,15 @@ String qb_inst_rpms_run_test() {
 String functional_rpms(String distro) {
     String rpms = "openmpi3 hwloc ndctl " +
                   "ior-hpc-cart-4-daos-0 " +
-                  "romio-tests-cart-4-daos-0 hdf5-mpich2-tests-daos-0 " +
+                  "romio-tests-cart-4-daos-0 " +
+                  "hdf5-mpich2-tests-daos-0 " +
+                  "hdf5-openmpi3-tests-daos-0 " +
                   "testmpio-cart-4-daos-0 fio " +
-                  "mpi4py-tests-cart-4-daos-0 MACSio"
+                  "mpi4py-tests-cart-4-daos-0 " +
+                  "MACSio-mpich2-daos-0 " +
+                  "MACSio-openmpi3-daos-0 " +
+                  "hdf5-vol-daos-mpich2-tests-daos-0 " +
+                  "hdf5-vol-daos-openmpi3-tests-daos-0"
```

Comment:
Let's make all of these (and the other test tools listed here, if you wll) dependencies (`Requires:`) of the _daos-tests_ subpackage.

Reply:
While I agree the daos test RPM should require these RPMs to function, we may want to consider separating out the daos test RPM into separate packages based upon some of these requirements.  In any case I think it is a discussion that can be had independently of this PR.

Reply:
So, while I agree that a separation of _daos-tests_ into some smaller packages might be a good idea in the future, I'd still like to stop the creeping tech-debt of this `functional_rpms` facility in `Jenkinsfile`.  Regardless of whether these functional test dependencies go into the _daos-tests_ RPM or are added to `functional_rpms` the result being the big dependency list is the same.  Moving it to the _daos-tests_ RPM just makes it explicit instead of implicit and causing people to have to go hunting for _daos-tests_' dependencies in another file.

Given that this PR is proposing to grow the tech-debt/implicit list, it seems like a good place to make this move.

Reply:
I've created https://jira.hpdd.intel.com/browse/DAOS-5544 to handle this change so it can be addressed independently.  I believe @mjean308 voiced some concerns as to how this would impact the use of RPMs outside of CI.